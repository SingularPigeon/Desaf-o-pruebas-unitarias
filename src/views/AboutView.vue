<template>
  <h1>About us</h1>
  <div>
    <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Iure, iste, mollitia adipisci neque harum eos expedita perferendis quo vero, repudiandae voluptatum molestiae officiis. Recusandae ex animi cupiditate exercitationem. Ipsam, voluptate.</p>
  </div>
</template>
