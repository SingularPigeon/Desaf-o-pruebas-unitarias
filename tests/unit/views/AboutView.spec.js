import { mount } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'

import AboutView from '@/views/AboutView.vue'


describe('AboutView', () => { 
  let routerPrueba
  
  beforeEach(async () => {
    routerPrueba = createRouter({
      history: createWebHistory(),
      routes: [{
        path: '/about',
        name: 'about',
        component: AboutView
      }],
    })
    routerPrueba.push('/about')
    await routerPrueba.isReady()
  })
  it('El HTML se renderiza correctamente', () => {
    const wrapper = mount(AboutView, {
      global: {
        plugins: [routerPrueba]
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  
  test('Probando la existencia la vista AboutView', async () => {
    const wrapper = mount(AboutView, {
      global: {
        plugins: [routerPrueba]
      }
    })
    expect(wrapper.findComponent(AboutView).exists()).toBe(true)
  })  

  test('Tiene un título "About us"', async () => {
    const wrapper = mount(AboutView, {
      global: {
        plugins: [routerPrueba]
      }
    })
    const h1 = wrapper.find('h1')
    expect(h1.text()).toBe('About us')
  })

  test('Tiene un parráfo "Lorem..."', async () => {
    const wrapper = mount(AboutView, {
      global: {
        plugins: [routerPrueba]
      }
    })
    const p = wrapper.find('p')

  expect(p.exists()).toBe(true)

  expect(p.text()).toContain('Lorem ipsum dolor sit amet')
  })
})
