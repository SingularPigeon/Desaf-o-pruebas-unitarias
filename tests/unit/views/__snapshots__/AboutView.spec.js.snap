// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutView El HTML se renderiza correctamente 1`] = `
<h1>About us</h1>
<div>
  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Iure, iste, mollitia adipisci neque harum eos expedita perferendis quo vero, repudiandae voluptatum molestiae officiis. Recusandae ex animi cupiditate exercitationem. Ipsam, voluptate.</p>
</div>
`;
